OakCityV2/TableViewController.m

```diff
@@ -0,0 +1,216 @@
+    operation.responseSerializer = [AFJSONResponseSerializer serializer];
+    
+    //Parse JSON
+    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
+        self.jobs  = [NSMutableArray new];
+        NSArray *responseArray = (NSArray *)responseObject;
+        
+        for (NSDictionary *responseDict in responseArray) {
+            Job* job = [[Job alloc] initWithDictionary:responseDict];
+            [self.jobs addObject:job];
+        }
+        
+        if([self.jobs count] > 0) {
+            [self.tableView reloadData];
+        }
+        
+        //Alert on failure to fetch JSON
+        else {
+            UIAlertView* alert = [[UIAlertView alloc]
+                                  initWithTitle: @"Failed to retrieve data" message: nil delegate: self
+                                  cancelButtonTitle: @"cancel" otherButtonTitles: @"Retry", nil];
+            [alert show];
+        }
+    }
+     
+    //Upon failure
+    failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+        UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:@"Error"
+                                                             message:[error localizedDescription]
+                                                            delegate:nil
+                                                   cancelButtonTitle:@"Ok"
+                                                   otherButtonTitles:nil];
+        [alertView show];
+    }];
+    [operation start];
+    [sender endRefreshing];
+}
+
+- (NSInteger)numberOfSectionsInTableView:tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView
+ numberOfRowsInSection:(NSInteger)section {
+    return self.jobs.count;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView
+estimatedHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
+    return UITableViewAutomaticDimension;
+}
+
+//Fill TableView
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    DetailCell *cell = [tableView dequeueReusableCellWithIdentifier:@"DetailCell" forIndexPath:indexPath];
+    
+    Job *aJob = self.jobs[indexPath.row];
+    [cell.titleLabel setText:aJob.title];
+    [cell.companyLabel setText:aJob.company];
+    if (aJob.logo != (NSString *)[NSNull null]) {
+        [cell.logoImageView setImageWithURL:[NSURL URLWithString:aJob.logo] placeholderImage:[UIImage imageNamed:@"placeholder.jpg"]];
+    }
+    else {
+        [cell.logoImageView setImage:[UIImage imageNamed:@"placeholder.jpg"]];
+    }
+    return cell;
+}
+
+//Fill DetailView
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    if ([segue.identifier isEqualToString:@"Identifier"]) {
+        DetailViewController *DetailVC = segue.destinationViewController;
+        
+        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+        
+        Job *theJob = self.jobs[indexPath.row];
+
```

Comment:
So this is where you should pass the job object to the detailVC and all the rest of the code here goes away.

In a more complex app, you might use the detailVC in multiple places so by moving all these details into the detailVC, you get consistent handling instead of a lot of code duplication.  (DUPLICATION BAD!)

And that goes hand in hand with ["separation of concerns"] (https://en.wikipedia.org/wiki/Separation_of_concerns).  The TableViewController shouldn't care how the DetailViewController deals with the job object.  The TableViewController just need to make sure it passes the correct job down to the DetailViewController.